reciprocalspaceship/algorithms/intensity.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+
+import reciprocalspaceship as rs
+from reciprocalspaceship.decorators import inplace
+
+
+@inplace
+def compute_intensity_from_structurefactor(
+    ds,
+    F_key,
+    SigF_key,
+    output_columns=None,
+    inplace=False,
+):
+    """
+    Compute intensities (I) and uncertainty estimates (SigI) from structure
+    factor amplitudes (F) and their uncertainties (SigF) using error propagation
+
+    Intensity computed as I = SigF*SigF + F*F. Intensity error estimate
+    approximated as SigI = abs(2*F*SigF)
+
+    Parameters
+    ----------
+    ds : DataSet
+        Input DataSet containing columns with F_key and SigF_key labels
+    F_key : str
+        Column label for structure factor amplitudes
+    SigF_key : str
+        Column label for structure factor error estimates
+    output_columns : list or tuple of column names
+        Column labels to be added to ds for calculated I and
+        SigI, respectively. output_columns must have len=2.
+    inplace : bool
+        Whether to modify the DataSet in place or create a copy
+
+    Returns
+    -------
+    DataSet
+        DataSet with 2 additional columns corresponding to calculated
+        intensities and intensity error estimates
+
+    """
+
+    # Sanitize input or check for invalid reflections
+    # if dropna:
+    #     ds.dropna(subset=[F_key, SigF_key], inplace=True)
+    # else:
+    #     if ds[[F_key, SigF_key]].isna().to_numpy().any():
+    #         raise ValueError(
+    #             f"Input {ds.__class__.__name__} contains NaNs "
+    #             f"in columns '{F_key}' and/or '{SigF_key}'. "
+    #             f"Please fix these input values, or run with dropna=True"
+    #         )
```

Comment:
Was this commented codeblock intended to make it into the last commit?

Reply:
Ah rats, no that can go away for sure